lib/newrelic-infra.js

```diff
@@ -194,15 +194,16 @@ var collectMetrics = function nriCollectMetrics(rawMetrics, rules) {
 
         if (data.hasOwnProperty(entityId)) {
           if (!data[entityId].metrics.hasOwnProperty(eventType)) {
-            data[entityId].metrics[eventType] = {};
+            data[entityId].metrics[eventType] = [];
           }
-          Object.assign(data[entityId].metrics[eventType], metricFields);
+          data[entityId].metrics[eventType].push(metricFields);
         } else {
           data[entityId] = {
             entity: { name: entityName, type: entityType },
             metrics: {}
           };
-          data[entityId].metrics[eventType] = metricFields;
+          data[entityId].metrics[eventType] = [];
```

Comment:
FYI you could write this as:

```
data[entityId] = {
  entity: { name: entityName, type: entityType },
  metrics: {
    [eventType]: [metricFields]
  }
};
```

Reply:
Just two points:
- This bracket notation of the `[eventType]` seems to be EcmaScript 6... The rest of the integration seems to be written in ES5.
- Why `[metricFields]` has also square brackets? Wouldn't it put `metricFields` inside an array?

Reply:
First point, I didn't know it was ES6 :(

Second point: that's what you are doing, you are first creating and empty array and then doing a push onto that empty array.

```
data[entityId].metrics[eventType] = [];
data[entityId].metrics[eventType].push(metricFields) 
== 
data[entityId].metrics[eventType] = [metricFields]
```